From 6ea20068855e84dd3255fc4dcb67a831f7dcd916 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 15 Feb 2022 08:14:49 +1000
Subject: [PATCH 1/4] meson.build: use datadir, not sysconfdir, for the
 xorg.conf files

Mixup of the two, sysconfdir is /etc which is not what we want for a
normal install of the driver

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 0b4bfe9e..237a669c 100644
--- a/meson.build
+++ b/meson.build
@@ -9,7 +9,7 @@ wacom_version = meson.project_version().split('.')
 dir_man1	= get_option('prefix') / get_option('mandir') / 'man1'
 dir_man4	= get_option('prefix') / get_option('mandir') / 'man4'
 dir_bin 	= get_option('prefix') / get_option('bindir')
-dir_x11_config	= get_option('prefix') / get_option('sysconfdir') / 'X11' / 'xorg.conf.d'
+dir_x11_config	= get_option('prefix') / get_option('datadir') / 'X11' / 'xorg.conf.d'
 dir_pkgconf	= get_option('prefix') / get_option('libdir') / 'pkgconfig'
 dir_udev	= get_option('prefix') / 'lib' / 'udev'
 dir_systemd	= get_option('prefix') / 'lib' / 'systemd'

From 703a5577aa0c371e1dea51dd16e8699bf0e4dd92 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 15 Feb 2022 08:15:51 +1000
Subject: [PATCH 2/4] meson.build: install the SDK headers

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 meson.build | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/meson.build b/meson.build
index 237a669c..b3c7c676 100644
--- a/meson.build
+++ b/meson.build
@@ -161,6 +161,14 @@ configure_file(
 	install_dir: dir_pkgconf,
 )
 
+install_headers(
+	'include/Xwacom.h',
+	'include/wacom-properties.h',
+	'include/isdv4.h',
+	'include/wacom-util.h',
+	install_dir: dir_wacom_headers
+)
+
 # GWacom library
 # This is not a stable library interface
 

From 23e37235bfb484714043f45d35f1c11204af0c16 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 15 Feb 2022 08:51:46 +1000
Subject: [PATCH 3/4] wacom-record: silence a compiler warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/usr/include/glib-2.0/glib/glib-autocleanups.h:28:3: warning: ‘syspath’
may be used uninitialized in this function [-Wmaybe-uninitialized]
   28 |   g_free (*pp);

Can't actually happen but it's an easy fix to get rid of this warning.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 tools/wacom-record.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/wacom-record.c b/tools/wacom-record.c
index 4e2a9f42..eb7cb248 100644
--- a/tools/wacom-record.c
+++ b/tools/wacom-record.c
@@ -302,7 +302,7 @@ static char *get_device_name(const char *path)
 	struct udev *udev;
 	struct udev_device *dev, *parent;
 	g_autofree char *basename = g_path_get_basename(path);
-	g_autofree char *syspath;
+	g_autofree char *syspath = NULL;
 	char *name = NULL;
 
 	udev = udev_new();

From 16fb7fe71038dd79b8370f5196b826eab4815c33 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 15 Feb 2022 09:01:43 +1000
Subject: [PATCH 4/4] test: skip the wacom tests if we can't open the .so file

Distribution package builds that use -Wl,-z,relro -Wl,-z,now will
attempt to resolve all symbols on dlopen(). This is explicitly what we
do not want because we don't link in the whole X server.

It's a bit hard to override this for a single executable (my attempts
didn't go far so there may be some other flag that influences this too)
so for now just skip the whole test - this is better than failing the
test suite.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 test/wacom-tests.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/wacom-tests.c b/test/wacom-tests.c
index ddc9c84f..63ca22f3 100644
--- a/test/wacom-tests.c
+++ b/test/wacom-tests.c
@@ -31,7 +31,8 @@ int main(void) {
 
 	if (handle == NULL) {
 		fprintf(stderr, "Failed to open %s: %s\n", TESTDRV, dlerror());
-		return 1;
+		fprintf(stderr, "This test suite relies on dlopen(RTLD_LAZY) which may be disabled by your compiler/linker flags\n");
+		return 77;
 	}
 
 	func = dlsym(handle, TESTFUNC);
